lib/tools/android-manifest.js

```diff
@@ -218,23 +218,45 @@ manifestMethods.compileManifest = async function compileManifest (manifest, mani
     throw new Error('Cannot compile the manifest. The required platform does not exist (API level >= 17)');
   }
   const resultPath = `${manifest}.apk`;
-  const args = [
-    'package',
-    '-M', manifest,
-    '--rename-manifest-package', manifestPackage,
-    '--rename-instrumentation-target-package', targetPackage,
-    '-I', path.resolve(platformPath, 'android.jar'),
-    '-F', resultPath,
-    '-f',
-  ];
+  const androidJarPath = path.resolve(platformPath, 'android.jar');
+  if (await fs.exists(resultPath)) {
+    await fs.rimraf(resultPath);
+  }
   try {
+    await this.initAapt2();
+    // https://developer.android.com/studio/command-line/aapt2
+    const args = [
+      'link',
+      '-o', resultPath,
+      '--manifest', manifest,
+      '--rename-manifest-package', manifestPackage,
+      '--rename-instrumentation-target-package', targetPackage,
+      '-I', androidJarPath,
+      '-v',
+    ];
+    log.debug(`Compiling the manifest using ${this.binaries.aapt2} ${quote(args)}`);
```

Comment:
For clarity of the command, I think there should be single quotes around the whole command (like https://github.com/appium/appium-adb/blob/c36f4eb1c6cf7672c28005b2edbdc36d7e6f4e37/lib/tools/system-calls.js#L316).

Reply:
changed